src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -102,12 +103,14 @@ public async Task<int> Monitor(CancellationToken ct, List<string> counter_list,
                 _console = console;
                 _interval = refreshInterval;
                 _renderer = new ConsoleWriter();
-                if (!BuildDiagnosticsClient())
+                if (!BuildDiagnosticsClient(port))
                 {
+                    CleanupPort(port);
```

Comment:
Having the CleanupPort call all over is error prone, could you make it in a finally block or wrap it in something that can be used with a using() block?